### PR TITLE
Feature/252 trading web api

### PR DESCRIPTION
**Pull request by vsatkov**
issue #252

### CODE REVIEW

File: HW7/Valeriy_Satkov/Trading/Traiding.WebAPIConsole/Controllers/ClientsController.cs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    public class ClientsController : ApiController
+    {
+        private readonly ClientsService clientsService;
+        private readonly BalancesService balancesService;
+        private readonly SalesService salesService;
+        private readonly ReportsService reportsService;
+
+        public ClientsController()
+        {
+            this.clientsService = new Container(new Models.DependencyInjection.TraidingRegistry()).GetInstance<ClientsService>();
```
Comment (chesdenis) [inline HW7/Valeriy_Satkov/Trading/Traiding.WebAPIConsole/Controllers/ClientsController.cs]: incorrect using of DI, you have to pass all services into ClientsController constructor as arguments
https://docs.microsoft.com/en-us/aspnet/core/mvc/controllers/dependency-injection?view=aspnetcore-2.2

File: HW7/Valeriy_Satkov/Trading/Traiding.ConsoleApp/DependencyInjection/TraidingRegistry.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿using StructureMap;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Traiding.ConsoleApp.DependencyInjection
+{
+    public class TraidingRegistry : Registry
```
Comment (chesdenis) [inline HW7/Valeriy_Satkov/Trading/Traiding.ConsoleApp/DependencyInjection/TraidingRegistry.cs]: why this class is empty?

File: HW7/Valeriy_Satkov/Trading/Traiding.ConsoleApp/StockExchange.cs [hunk 3]
```diff
@@ -0,0 +1,516 @@
...
+                Console.WriteLine(" 4. Print client balance zone by Client Id");
+                Console.WriteLine(" 5. Print client operations");
+                Console.WriteLine(" 6. Add a new share");
+                Console.WriteLine(" 7. Edit share info by Id");
+                Console.WriteLine(" 8. Remove share by Id");
+                Console.WriteLine(String.Empty);
+                Console.Write("Type the number or 'e' for exit and press Enter: ");
+
+                inputString = Console.ReadLine();
+                switch (inputString)
```
Comment (chesdenis) [inline HW7/Valeriy_Satkov/Trading/Traiding.ConsoleApp/StockExchange.cs]: please try to use pattern strategy here...

File: HW7/Valeriy_Satkov/Trading/Traiding.Core/Repositories/IBlockedSharesNumberTableRepository.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+﻿namespace Traiding.Core.Repositories
+{
+    using Traiding.Core.Models;
+
+    public interface IBlockedSharesNumberTableRepository
+    {
+        //bool Contains(BlockedSharesNumberEntity entity); // Compare by
```
Comment (chesdenis) [inline HW7/Valeriy_Satkov/Trading/Traiding.Core/Repositories/IBlockedSharesNumberTableRepository.cs]: why you have these commented blocks?

File: HW7/Valeriy_Satkov/Trading/Traiding.WebAPIConsole/Controllers/ValuesController.cs [hunk 5]
```diff
@@ -0,0 +1,39 @@
+﻿namespace Traiding.WebAPIConsole.Controllers
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http;
+    using System.Web.Http;
+
+    public class ValuesController : ApiController
```
Comment (chesdenis) [inline HW7/Valeriy_Satkov/Trading/Traiding.WebAPIConsole/Controllers/ValuesController.cs]: what is the purpose of this controller?

File: HW7/Valeriy_Satkov/Trading/Traiding.Core/Repositories/IBlockedSharesNumberTableRepository.cs (hunk 4 above)
Comment (vsatkov) [reply HW7/Valeriy_Satkov/Trading/Traiding.Core/Repositories/IBlockedSharesNumberTableRepository.cs]: This is no longer necessary. I cannot create the same 'BlockedSharesNumberEntity'

**Pull request merged**